src/formatDate.js

```diff
@@ -8,7 +8,42 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat.pop();
+  const newSeparator = toFormat.pop();
+  const dateParts = date.split(separator);
+  const newDate = ['', '', ''];
+
+  const formatMap = {
+    DD: 'день',
+    MM: 'місяць',
+    YYYY: 'рік',
+    YY: 'рік',
+  };
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    const format = fromFormat[i];
+    const partType = formatMap[format];
+    const partValue = dateParts[i];
+
+    let newIndex = -1;
```

Comment:
Initializing `newIndex` with `-1` is risky because if it's not updated correctly in the loop, it will cause an index out of bounds error when used. Ensure there's a check to handle the case where `newIndex` remains `-1` after the loop.